pkg/genericcomp/genericcomp.go

```diff
@@ -18,11 +19,21 @@ const GenericComponentSelectorLabel = "verrazzano.name"
 
 // NewDeployment constructs a deployment for a generic component.
 func NewDeployment(generic v1beta1v8o.VerrazzanoGenericComponent, bindingName string, namespace string, labels map[string]string) *appsv1.Deployment {
+	labels["app"] = generic.Name
+
+	_, targetPort := getPorts(&generic)
+
+	annotations := make(map[string]string)
+	annotations["prometheus.io/scrape"] = "true"
+	annotations["prometheus.io/port"] = fmt.Sprint(targetPort)
+	annotations["prometheus.io/path"] = "/actuator/prometheus"
```

Comment:
Does this depend on if the prometheus actuator is enabled in the genericComponent/application?

Reply:
Hey @shihchang 
Yes it does.  
Some info here
https://gitlab-odx.oracledx.com/verrazzano/springboot-app
and here
https://stackabuse.com/monitoring-spring-boot-apps-with-micrometer-prometheus-and-grafana/

Basically all custom springboot endpoints hang off of actuator
There is also a dependency that needs to be added for build
Plan to add a configurable metrics endpoint after this gets checked in